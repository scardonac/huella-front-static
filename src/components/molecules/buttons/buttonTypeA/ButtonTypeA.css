/* button.ButtonTypeA:hover{
    background-color: red ;
} */