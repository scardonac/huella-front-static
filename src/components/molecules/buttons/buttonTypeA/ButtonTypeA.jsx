import { useState } from 'react'

export const ButtonTypeA = ({
    text = "defaultText",
    action = null,
    width = "264px",
    bgColor = "#f7f9f7",
    bgHvColor = "#FFFFFF",
    txColor = "#627173",
    bdType = "solid",
    bdWidth = "1px",
    bdColor = "#627173",
    fontSize = "18px",
    submitBtn = false,
    centrado = false,
}) => {

    const buttonStyle = {
        border: `${bdWidth} ${bdType} ${bdColor}`,
        backgroundColor: bgColor,
        color: txColor,
        width: width,
        fontSize: fontSize,
        padding: "7px 0px",
        borderRadius: "10px",
        fontWeight: "bold",
        margin: centrado ? "0 auto" : "inehirt",
        height: "fit-content",
    };

    const hoverStyle = {
        backgroundColor: bgHvColor
    };

    const [isHovered, setIsHovered] = useState(false);

    const handleMouseEnter = () => {
        setIsHovered(true);
    };

    const handleMouseLeave = () => {
        setIsHovered(false);
    };

    return (
        <button
            className='ButtonTypeA'
            onMouseEnter={handleMouseEnter}
            onMouseLeave={handleMouseLeave}
            style={isHovered ? { ...buttonStyle, ...hoverStyle } : buttonStyle}
            onClick={action}
            type={submitBtn ? "submit" : "button"}
        >{text}
        </button>
    )
}