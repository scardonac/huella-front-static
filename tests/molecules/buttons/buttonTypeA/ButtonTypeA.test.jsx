//Testing-library
import { fireEvent, render, screen } from "@testing-library/react";
//utils-test-dom
import { act } from "react-dom/test-utils";
//Component
import { ButtonTypeA } from "../../../../src/components/molecules/buttons/buttonTypeA/ButtonTypeA";

describe('Pruebas en <ButtonTypeA />', () => {

    test('Debe renderizarse correctamente', () => {
        const { baseElement } = render(<ButtonTypeA />);
        expect(baseElement).toMatchSnapshot();
    });

    test('Debe renderizarse correctamente con estilos', () => {
        render(<ButtonTypeA style="test" />);

        const fnButton = screen.getByTestId('button_type_a');

        act(() => {
            fireEvent.click(fnButton);
        });

    });

    test('Debe renderizarse correctamente con estilos centrado', () => {
        render(<ButtonTypeA centrado />);

        const fnButton = screen.getByTestId('button_type_a');

        act(() => {
            fireEvent.click(fnButton);
        });

    });

    test('Debe renderizarse correctamente con estilos centrado y submit', () => {
        render(<ButtonTypeA centrado={true} submitBtn={() => { }} />);

        const fnButton = screen.getByTestId('button_type_a');

        act(() => {
            fireEvent.click(fnButton);
        });

    });

    test('Debe renderizarse correctamente con onMouseLeave y onMouseEnter', () => {
        render(<ButtonTypeA />);

        const fnButton = screen.getByTestId('button_type_a');

        act(() => {
            fireEvent.mouseEnter(fnButton);
        });

        act(() => {
            fireEvent.mouseLeave(fnButton);
        });

    });

});