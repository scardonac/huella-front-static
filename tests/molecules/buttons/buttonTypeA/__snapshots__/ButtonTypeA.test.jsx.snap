// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ButtonTypeA /> Debe renderizarse correctamente 1`] = `
<body>
  <div>
    <button
      class="ButtonTypeA"
      data-testid="button_type_a"
      id="button_type_a"
      style="border: 1px solid #627173; background-color: rgb(247, 249, 247); color: rgb(98, 113, 115); width: 264px; font-size: 18px; padding: 7px 0px; border-radius: 10px; font-weight: bold;"
      type="button"
    >
      defaultText
    </button>
  </div>
</body>
`;
